### PR TITLE
Add rosserial support for mbed platforms

### PR DESCRIPTION
**Pull request by garyservin**
Working implementation of rosserial for mbed platform.
- Based on work from [nucho](https://developer.mbed.org/users/nucho/code/rosserial_mbed_lib/)
- Tested on LPC1768, KL25Z and Nucleo F401RE platforms
- Tested most of the examples (I don't have some of the additional hardware required)
- Works with the ofline compiler from [adamgreen](https://github.com/adamgreen/gcc4mbed)
  - It should work with the online compiler but haven't found an easy way to upload the library to the compiler
- ~~Examples and tests are not installed in the same manner as with rosserial_arduino since the compiler builds **everything** on the subfolders of the project. See https://github.com/adamgreen/gcc4mbed/issues/37 for reference~~ This has been fixed in the latest release of gcc4mbed
- Provided a simple readme with instructions on how to install the library

Suggestions are welcome!

### CODE REVIEW

File: rosserial_mbed/src/examples/ADC/makefile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+PROJECT         := rosserial_mbed_ADC
+DEVICES         := LPC1768
+#GCC4MBED_DIR    := /home/gary/devel/mbed/gcc4mbed
```
Comment (mikepurvis) [inline rosserial_mbed/src/examples/ADC/makefile]: Can this be an env var?

Comment (garyservin) [inline rosserial_mbed/src/examples/ADC/makefile]: done!

**Comment by sorki** [comment]
Got this working on KL25Z. Tried HelloWorld and ADC examples, both work nicely.

Good job!

**Comment by garyservin** [comment]
@mikepurvis any more comments on this PR? Should I change anything else so it can be accepted? Thanks!

**Comment by mikepurvis** [comment]
This looks really great. Sorry for the ridiculous delay in getting it merged. Could you just comment briefly on the `GCC4MBED_DIR` variable? Is there anyone packaging this software for Ubuntu? How much is in it beyond the [standard gnuarm](https://launchpad.net/gcc-arm-embedded) stuff that's already easily installable? Alternatively, and way that this location could be autodiscovered using the extras script? Looks like you maybe tried this and gave up on it?

Also, should we put BSD headers on the examples?

**Comment by mikepurvis** [comment]
@garyservin ping

**Comment by garyservin** [comment]
Hi Mike, the GCC4MBED_DIR define where the mbed SDK and makefiles for the different platforms are (external/mbed and build dirs on https://github.com/adamgreen/gcc4mbed).

No one is packaging this unfortunately. I've tried using the standard gnuarm and it works but then you need to add the mbed SDK and the makefiles I've mentioned. And no, I haven't tried to autodiscover this, since it depends on where you've cloned the gcc4mbed repo.

The rosserial_arduino example don't have headers, should we still add them to these examples?

**Comment by mikepurvis** [comment]
Aight, let's get this in. Good to go from your side?

**Comment by garyservin** [comment]
yes, good to go!

**Pull request merged**